go-controller/pkg/ovn/secondary_layer2_network_controller_test.go

```diff
@@ -136,6 +137,108 @@ var _ = Describe("OVN Multi-Homed pod operations for layer2 network", func() {
 			dummyPrimaryLayer2UserDefinedNetwork("100.200.0.0/16"),
 		),
 	)
+
+	table.DescribeTable(
+		"the gateway is properly cleaned up",
+		func(netInfo secondaryNetInfo, testConfig testConfiguration) {
+			podInfo := dummyTestPod(ns, netInfo)
+			if testConfig.configToOverride != nil {
+				config.OVNKubernetesFeature = *testConfig.configToOverride
+			}
+			app.Action = func(ctx *cli.Context) error {
+				netConf := netInfo.netconf()
+				networkConfig, err := util.NewNetInfo(netConf)
+				Expect(err).NotTo(HaveOccurred())
+
+				nad, err := newNetworkAttachmentDefinition(
+					ns,
+					nadName,
+					*netConf,
+				)
+				Expect(err).NotTo(HaveOccurred())
+
+				const nodeIPv4CIDR = "192.168.126.202/24"
+				testNode, err := newNodeWithSecondaryNets(nodeName, nodeIPv4CIDR, netInfo)
+				Expect(err).NotTo(HaveOccurred())
+
+				gwConfig, err := util.ParseNodeL3GatewayAnnotation(testNode)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(gwConfig.NextHops).NotTo(BeEmpty())
+
+				if netInfo.isPrimary {
+					gwConfig, err := util.ParseNodeL3GatewayAnnotation(testNode)
+					Expect(err).NotTo(HaveOccurred())
+					initialDB.NBData = append(
+						initialDB.NBData,
+						expectedLayer2EgressEntities(networkConfig, *gwConfig, nodeName)...)
+				}
+
+				fakeOvn.startWithDBSetup(
+					initialDB,
+					&v1.NamespaceList{
+						Items: []v1.Namespace{
+							*newNamespace(ns),
+						},
+					},
+					&v1.NodeList{
+						Items: []v1.Node{*testNode},
+					},
+					&v1.PodList{
+						Items: []v1.Pod{
+							*newMultiHomedPod(podInfo.namespace, podInfo.podName, podInfo.nodeName, podInfo.podIP, netInfo),
+						},
+					},
+					&nadapi.NetworkAttachmentDefinitionList{
+						Items: []nadapi.NetworkAttachmentDefinition{*nad},
+					},
+				)
+
+				Expect(netInfo.setupOVNDependencies(&initialDB)).To(Succeed())
+
+				podInfo.populateLogicalSwitchCache(fakeOvn)
+
+				// pod exists, networks annotations don't
+				pod, err := fakeOvn.fakeClient.KubeClient.CoreV1().Pods(podInfo.namespace).Get(context.Background(), podInfo.podName, metav1.GetOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				_, ok := pod.Annotations[util.OvnPodAnnotationName]
+				Expect(ok).To(BeFalse())
+
+				Expect(fakeOvn.controller.WatchNamespaces()).To(Succeed())
+				Expect(fakeOvn.controller.WatchPods()).To(Succeed())
+				secondaryNetController, ok := fakeOvn.secondaryControllers[secondaryNetworkName]
+				Expect(ok).To(BeTrue())
+
+				secondaryNetController.bnc.ovnClusterLRPToJoinIfAddrs = dummyJoinIPs()
+				podInfo.populateSecondaryNetworkLogicalSwitchCache(fakeOvn, secondaryNetController)
+				Expect(secondaryNetController.bnc.WatchNodes()).To(Succeed())
+				Expect(secondaryNetController.bnc.WatchPods()).To(Succeed())
+
+				Expect(fakeOvn.fakeClient.KubeClient.CoreV1().Pods(pod.Namespace).Delete(context.Background(), pod.Name, metav1.DeleteOptions{})).To(Succeed())
```

Comment:
some ginkgo.By steps would be good in here so we can see easier what happens in the test case log

Reply:
Done.